src/sentry/api/endpoints/organization_discover_query.py

```diff
@@ -88,27 +87,28 @@ def __init__(self, *args, **kwargs):
     def validate(self, data):
         data['arrayjoin'] = self.arrayjoin
 
-        return data
-
-    def validate_range(self, attrs, source):
-        has_start = bool(attrs.get('start'))
-        has_end = bool(attrs.get('end'))
-        has_range = bool(attrs.get('range'))
-
-        if has_start != has_end or has_range == has_start:
+        if data.get('start') or data.get('end'):
+            if not all([data.get('start'), data.get('end')]):
+                raise InvalidParams('start and end are both required')
```

Comment:
You probably want `raise serializers.ValidationError("...")` here so we get a 4xx instead of 5xx

Reply:
yeah this was a copypasta, will fix